frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -244,22 +244,18 @@ private Notification getSkippedBranchesNotification(IGitMacheteRepositorySnapsho
         }));
     notification.addAction(NotificationAction.createSimple(
         () -> getString("action.GitMachete.OpenMacheteFileAction.description"), () -> {
-          val actionEvent = getAnActionEvent();
+          val actionEvent = createAnActionEvent();
           ActionManager.getInstance().getAction(ACTION_OPEN_MACHETE_FILE).actionPerformed(actionEvent);
         }));
     return notification;
   }
 
   private void slideOutBranch(IGitMacheteRepositorySnapshot repositorySnapshot, GitRepository gitRepository) {
-    IBranchLayout branchLayout = repositorySnapshot.getBranchLayout().getOrNull();
-    if (branchLayout == null) {
-      // todo: handle
-      return;
-    }
-
-    IBranchLayout newBranchLayout = branchLayout;
+    IBranchLayout newBranchLayout = repositorySnapshot.getBranchLayout();
     for (val branchName : repositorySnapshot.getSkippedBranchNames()) {
-      newBranchLayout = newBranchLayout.slideOut(branchName);
+      do {
+        newBranchLayout = newBranchLayout.slideOut(branchName);
+      } while (newBranchLayout.findEntryByName(branchName).isDefined());
```

Comment:
Hmm why would the entry for `branchName` be defined in `newBranchLayout` after a slide out? 🤔

Reply:
slideOut slides out only the first occurrence

Reply:
I think you could sort out #695 here in one go, by changing `BranchLayout` slightly:

```java
  @Override
  public IBranchLayout slideOut(Predicate<IBranchLayoutEntry> shouldBeSlidOut) {
    // TODO (#695): disable slide out for duplicated branches
    return new BranchLayout(rootEntries.flatMap(rootEntry -> slideOut(rootEntry, shouldBeSlidOut)));
  }

  private List<IBranchLayoutEntry> slideOut(IBranchLayoutEntry entry, Predicate<IBranchLayoutEntry> shouldBeSlidOut) {
    val children = entry.getChildren();
    if (shouldBeSlidOut.test(entry)) {
      return children;
    } else {
      return List.of(entry.withChildren(children.flatMap(child -> slideOut(child, shouldBeSlidOut))));
    }
  }
```

Reply:
(+ a regression test would be of gold's value)